### PR TITLE
Added required description comment in binary_search.cpp

### PR DESCRIPTION
**Pull request by varshika03**
Review it so that I can add more.

### CODE REVIEW

**Comment by quantum-pie** [comment]
I think space complexity is O(1) because you don't need additional memory

File: Searching/BinarySearch.cpp [hunk 1]
```diff
@@ -1,7 +1,13 @@
 /*
-    Binary search:
-    A search algorithm that finds the position of a target value
-    within a sorted array
+    <Binary Search>:
+    <A search algorithm that finds the position of a target value
+    within a sorted array>
+
+    <Time complexity>:
+    O(logn), where <n is the number of elements in the array>
```
Comment (faheel) [inline Searching/BinarySearch.cpp]: I suggest writing `O(logn)` as either `O(logN)`, `O(log(n))` or `O(log(N))` for more clarity.

`O(log(N))` would be the best way IMO.

**Review by faheel — CHANGES_REQUESTED** [review]
Remove all the angle brackets (`<` & `>`). They are used in the format just for showing that the enclosing text is a description of the text, and not the actual text.

File: Searching/BinarySearch.cpp [hunk 2]
```diff
@@ -1,7 +1,13 @@
...
-    within a sorted array
+    <Binary Search>:
+    <A search algorithm that finds the position of a target value
+    within a sorted array>
+
+    <Time complexity>:
+    O(logn), where <n is the number of elements in the array>    
+    
+    <Space complexity>:
+    O(n), where <n is the number of elements in the array>
```
Comment (faheel) [inline Searching/BinarySearch.cpp]: It's actually `O(1)` as only a constant amount of **extra memory** is required.

Comment (faheel) [inline Searching/BinarySearch.cpp]: It's actually `O(1)` as only a constant amount of **extra memory** is required.

**Review by faheel — CHANGES_REQUESTED** [review]

**Review by faheel — CHANGES_REQUESTED** [review]
Newlines please! As in this:
```C++
/*
    <Name of algorithm>:
    <Brief description>

    Time complexity:
    O(...), where <description of variable(s)>    
    
    Space complexity:
    O(...), where <description of variable(s)>
*/
```

And again, I suggest writing `O(logn)` as either `O(logN)`, `O(log(n))` or `O(log(N))` for more clarity.
I think `O(log(N))` would be the best way.

**Comment by faheel** [comment]
Closing this since there hasn't been any activity for a long time.

**Pull request closed without merging**